### PR TITLE
chore: Refactoring OpenShift Initial user reconciler

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Refactoring OpenShift Initial user reconciler

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

N/A

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: api/v1/checluster_types.go [hunk 1]
```diff
@@ -781,3 +781,11 @@ func (c *CheCluster) IsImagePullerImagesEmpty() bool {
 func (c *CheCluster) IsInternalClusterSVCNamesEnabled() bool {
 	return c.Spec.Server.DisableInternalClusterSVCNames == nil || !*c.Spec.Server.DisableInternalClusterSVCNames
 }
+
+func (c *CheCluster) IsOpenShiftOAuthUser() bool {
```
Comment (sleshchenko) [inline api/v1/checluster_types.go:785]: AFAIK Che now supports kubeadmin just fine, should not we drop that ability at all?

Comment (tolusha) [reply api/v1/checluster_types.go:785]: Only when devworkspace is enabled

File: pkg/deploy/openshift-oauth/openshiftoauth_user.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+func (oou *OpenShiftOAuthUser) Finalize(ctx *deploy.DeployContext) error {
+	if util.IsOpenShift4 {
+		return oou.Delete(ctx)
+	}
+
+	return nil
+}
+
+// Creates new htpasswd provider with initial user with Che flavor name
+// if Openshift cluster hasn't got identity providers, otherwise do nothing.
```
Comment (mmorhun) [inline pkg/deploy/openshift-oauth/openshiftoauth_user.go:107]:
```suggestion
// if Openshift cluster hasn't got identity providers, does nothing.
```

File: pkg/deploy/reconcile_manager.go [hunk 3]
```diff
@@ -21,9 +21,9 @@ import (
 
 type Reconcilable interface {
 	// Reconcile object.
-	Reconcile(ctx *DeployContext) (result reconcile.Result, done bool, err error)
+	Reconcile(ctx *DeployContext) (reconcile.Result, bool, error)
```
Comment (mmorhun) [inline pkg/deploy/reconcile_manager.go:24]: Why did you remove variables names? They show purpose of each arg and even if 2 of 3 are obvious, I'd keep it for the `done` one.
